README.md

```diff
@@ -0,0 +1,66 @@
+# slog, a back-end agnostic interface for structured logs
+
+[![Go Reference](https://pkg.go.dev/badge/github.com/darvaza-proxy/slog.svg)](https://pkg.go.dev/github.com/darvaza-proxy/slog)
+
+[`slog.Logger`](#interface) provides a simple standardised interface for structured logs for libraries. It supports [six log levels](#log-levels) and [fields](#fields) with unique string labels (keys).
+
+## Interface
+Every method of this interface, with the exception of [`Print()`](#print), returns a `slog.Logger` so it can be daisy chained when composing a log entry.
+A log entry begins setting the [level](#log-levels) followed by optional addition of [fields](#fields) and a [call stack](#call-stack) and ends with a message calling a [Print](#print) method.
+Based on the specified [level](#log-levels) an entry could be [enabled or disabled](#enabled), and calls to methods on disabled entries will cause no action unless it's used to create a new entry with a [level](#log-levels) that is enabled.
+
+## Log Levels
+An `slog.Logger` entry can be of one of 6 levels, of which Fatal is expected to end the execution just like the standard `log.Fatal()`right after adding the log entry, and Panic to raise a recoverable panic like `log.Panic()`.
+
+ 1. Debug
+ 2. Info
+ 3. Warn
+ 4. Error
+ 5. Fatal
+ 6. Panic
+
+New log entries can be created by calling the named shortcut methods (`Debug()`, `Info()`, `Warn()`, `Error()`, `Fatal()`, and `Panic()`) or via `WithLevel(level)`.
+
+
+## Enabled
+A log entry is considered Enabled if the back-end would actually log entries of the specified level.
+When a logger is not `Enabled()` string formatting operations or fields and stack commands are not performed, but when it's useful to know if certain level is *Enabled* before deciding how much detail we want to log. For this purpose `WithEnabled()` can be used like:
+```go
+if log, ok := logger.Debug().WithEnabled(); ok {
+	log.WithField("request", req).Print("Let's write detailed debug stuff")
+} else if log, ok := logger.Info().WithEnabled(); ok {
+	log.Print("Let's write info stuff instead")
+}
+```
+
+Regardless the _Enabled_ state, Fatal and Panic levels are always expected to exit/panic accordingly.
```

Comment:
Logs of Fatal and Panic level are expected to exit/panic regardless of the _Enabled_ state.